labgrid/remote/client.py

```diff
@@ -693,29 +693,34 @@ def power(self):
         action = self.args.action
         delay = self.args.delay
         target = self._get_target(place)
+        from ..protocol.powerprotocol import PowerProtocol
         from ..driver.powerdriver import NetworkPowerDriver, PDUDaemonDriver, USBPowerDriver
         from ..resource.power import NetworkPowerPort, PDUDaemonPort
         from ..resource.remote import NetworkUSBPowerPort
+
         drv = None
-        for resource in target.resources:
-            if isinstance(resource, NetworkPowerPort):
-                try:
-                    drv = target.get_driver(NetworkPowerDriver)
-                except NoDriverFoundError:
-                    drv = NetworkPowerDriver(target, name=None, delay=delay)
-                break
-            elif isinstance(resource, NetworkUSBPowerPort):
-                try:
-                    drv = target.get_driver(USBPowerDriver)
-                except NoDriverFoundError:
-                    drv = USBPowerDriver(target, name=None, delay=delay)
-                break
-            elif isinstance(resource, PDUDaemonPort):
-                try:
-                    drv = target.get_driver(PDUDaemonDriver)
-                except NoDriverFoundError:
-                    drv = PDUDaemonDriver(target, name=None, delay=int(delay))
-                break
+        try:
+            target.get_driver(PowerProtocol)
```

Comment:
Why don't you use the name as a string to avoid the import?

Reply:
Because at this point, no class has been registered in case no environment file was given on the command line. I don't remember why we only allow already bound drivers to be resolved by string.